source/leuven/services/openondemand.rst

```diff
@@ -191,136 +241,324 @@ of using any of the other partitions. There are however some limitations on the
+
+- Pick a specific 'Toolchain and Python versions', e.g. '2023a and ``Python/3.11.3-GCCcore-12.3.0``'.
+- Choose a specific architecture, e.g. Sapphire Rapids nodes on wICE
+- Start an :ref:`Interactive Shell<interactive_shell>` on the targeted architecture,
+  and execute the following to create an environment called ``<kernel_name>``; you may choose a ``<kernel_name>``
+  which represents your project scope:
+
+  .. code-block :: bash
+
+     TOOLCHAIN='2023a'
+     DIR_VENV=${VSC_DATA}/apps/venv/${VSC_OS_LOCAL}/${VSC_ARCH_LOCAL}${VSC_ARCH_SUFFIX}/${TOOLCHAIN}/<kernel_name>
+     mkdir -p ${DIR_VENV}
+     # the line below is needed if you use 'Interactive Shell' app
+     module use /apps/leuven/${VSC_OS_LOCAL}/${VSC_ARCH_LOCAL}${VSC_ARCH_SUFFIX}/${TOOLCHAIN}/modules/all
+     module load Python/3.11.3-GCCcore-12.3.0
+     python -m venv ${DIR_VENV}
+     source ${DIR_VENV}/bin/activate
+     pip install --prefix=${DIR_VENV} ipykernel <additional packages>
+     # note that below, the "--env ..." argument is not needed
+     python -m ipykernel install --user --name <kernel_name> --display-name <kernel_name>
+
+- On the JupyterLab form, choose a partition to your liking and select the same toolchain as above.
+- Once you connect to your session, your new ``<kernel_name>`` is ready to be used.
+  To verify your setup, you can execute ``import sys; sys.executable`` in your notebook,
+  and the resulting path shall point at ``DIR_VENV`` where you installed your virtual environment.
+
+**Remarks:**
+
+- If one your package installation steps involves compiling source code, then you might only be able to
+  use your virtual environment on the same architecture where the compilation was carried out. 
+  This remark does not apply to typical ``pip`` usage where precompiled 'wheels' get downloaded and installed
+  and which can therefore be used on any architecture.
+
+Conda environments for R
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+For R, you need both the ``jupyter_client`` and the ``irkernel`` Conda packages installed.
+With the following command you can create the kernel::
 
       Rscript -e 'IRkernel::installspec(name="<env_name>", displayname="<kernel_name>")'
 
-Once the kernel is created, you will see it in the 'Launcher' menu. You can now start working in your own customized environment.
+Once the kernel is created, you will see it in the 'Launcher' menu.
+You can now start working in your own customized environment.
 
-For more general information concerning JupyterLab, go to their `official documentation <https://docs.jupyter.org/en/latest/>`__.
+For more general information, please refer to the `official JupyterLab documentation`_.
 
 **Remarks:**
 
-- The start location is `$VSC_DATA`
-- At the moment, we do not support installing extensions in Jupyter Lab
+- The top-level notebook directory is by default ``$VSC_DATA``.
+- At the moment, we do not support installing extensions in JupyterLab.
+
+.. note::
```

Comment:
In 1c9bf78, I moved up the remarks to the beginning of the JupyterLab section, before the "Pure Python environments".

However, the scope of the 'Note' box applies to all Python/R Conda/venv kernels. 
The problem with (our) restructured text is that the subsectioning is not fully clear, eventhough we have consistently used correct underline characters for the (sub)section titles. As a result, the 'Note' box appears below the R Conda section, but in practice it shall belong to all kernel-driven envs.

Reply:
Well, you could still move the Note up to the beginning of the section, right? Then that's also where it will appear. But on second thought it is probably best to not move that Note to the top (only the remarks, as you just did), since it's too detailed to be placed at the beginning.

How about this:
First it would indeed be great if the (sub, subsub, ...)section titles on this page would get numbered. I don't think that really has to do with how the titles are underlined, but how the page's [TOC tree](https://documatt.com/restructuredtext-reference/element/toctree.html) is set up. Could you look into that?
Then it would just be a matter of adding another `Remarks` subsection after the R Conda subsection to include additional notes (such as this one) and the numbering will then make clear that this applies to the whole JupyterLab section.

Reply:
I tried out enumerating the (sub)sections, and it did not turn out a success, because:
- the 'Note' box did still sit below the same subsection for 'Conda environment for R', and it did not look like belonging to the whole JupyterLab section
- the section enumeration does not impose indenting the subsections, or making them recognizable. So, the ambiguity stayed.

Instead, I created a new sub(sub?)section under the JupyterLab section, and move the 'Note' block up inside this one. You may see it in 1e7a459.

Reply:
@MaximeVdB what about the changes in c24eff6? The sections are only numbered, without (a long) table of content.

Reply:
Yes that looks pretty good (the only downside being that the page title itself also gets numbered; but it looks like there is not an easy way to avoid that, so for the time being this is OK for me).